tests/api_validation_test.go

```diff
@@ -113,6 +113,37 @@ var _ = Describe("[rfe_id:1130][crit:medium][posneg:negative][vendor:cnv-qe@redh
 				table.Entry("[test_id:3928]fail with empty VDDK source thumbprint", "vddk", validURL, "secret", "uuid", "backingfile", ""),
 			)
 
+			table.DescribeTable("with DataVolume sourceRef validation should", func(kind, namespace, name string) {
+				By("Reading yaml file from: " + datavolumeTestFile)
+				err := yamlFiletoStruct(datavolumeTestFile, &dv)
+				Expect(err).ToNot(HaveOccurred())
+
+				delete(dv["spec"].(map[string]interface{}), "source")
+				dv["spec"].(map[string]interface{})["sourceRef"] = map[string]interface{}{
+					"kind":      kind,
+					"namespace": &namespace,
+					"name":      name}
+
+				err = structToYamlFile(destinationFile, dv)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Verifying kubectl create")
+				Eventually(func() bool {
+					out, err := RunKubectlCommand(f, "create", "-f", destinationFile, "-n", f.Namespace.Name)
+					By("out: " + out)
+					if err != nil {
+						return true
+					}
+					return false
+				}, timeout, pollingInterval).Should(BeTrue())
+
+			},
+				table.Entry("[test_id:XXXX]fail with empty kind", "", "test", "test-pvc"),
+				table.Entry("[test_id:XXXX]fail with unsupported kind", "no-such-kind", "test", "test-pvc"),
+				table.Entry("[test_id:XXXX]fail with empty sourceRef name", "DataSource", "test", ""),
+				table.Entry("[test_id:XXXX]fail with sourceRef DataSource doesn't exist", "DataSource", "test", "test-pvc"),
+			)
```

Comment:
table.Entry("[test_id:6780]fail with empty kind", "", "test", "test-pvc"),
table.Entry("[test_id:6781]fail with unsupported kind", "no-such-kind", "test", "test-pvc"),
table.Entry("[test_id:6782]fail with empty sourceRef name", "DataSource", "test", ""),
table.Entry("[test_id:6783]fail with sourceRef DataSource doesn't exist", "DataSource", "test", "test-pvc"),

Reply:
Thanks Yan, but you missed:
`[test_id:XXXX]fail with both source and sourceRef`